kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,6 +1351,86 @@ class ResidualBasedBlockBuilderAndSolver
         }
     }
 
+    /**
+     * @brief This method returns the scale norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @param rCurrentProcessInfo The current process info instance
+     * @param rScaleVariable The scale variable considered
+     * @return The scale norm
+     */
+    double GetScaleNorm(
+        TSystemMatrixType& rA,
+        const ProcessInfo& rCurrentProcessInfo,
+        const Variable<double>& rScaleVariable
+        )
+    {
+        if (mOptions.Is(SCALE_DIAGONAL) ) {
+            if (rCurrentProcessInfo.Has(rScaleVariable)) {
+                return rCurrentProcessInfo[rScaleVariable];

```

Comment:
why do you want to get it from the processinfo?

Reply:
In order to avoid recompute (or to manually set the value if it is of interest)

Reply:
but then, why don't we pass it directly in the parameters?

Reply:
Essentially what i propose here would be to pass in the parameters an unique option like

"diagonal_values_for_dirichlet_dofs" (very open to selecting a better name)

with possible values like

* "use_max_diagonal" <-- default
* "use_average_diagonal" 
* 1000.0 <--- here the user could write a number...

Reply:
please take a look at this...

Reply:
@philbucher see here